File: wear/src/main/java/cz/johrusk/showsmscode/BootCompletedReceiver.java
package cz.johrusk.showsmscode;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Created by Pepa on 19.06.2016.
 */

public class BootCompletedReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if ("android.intent.action.BOOT_COMPLETED".equals(intent.getAction())) {
            Intent pushIntent = new Intent(context, DataLayerListenerService.class);
            context.startService(pushIntent);
        }
    }
}

File: wear/src/main/java/cz/johrusk/showsmscode/BootCompletedReceiver.kt
package cz.johrusk.showsmscode

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * Broadcast receiver which start Datalayerlistener when is Android Wear started
 *
 * @author Josef Hruska (pepa.hruska@gmail.com)
 */

class BootCompletedReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if ("android.intent.action.BOOT_COMPLETED" == intent.action) {
            val pushIntent = Intent(context, DataLayerListenerService::class.java)
            context.startService(pushIntent)
        }
    }
}
